$MirOS: ports/mailnews/pine/patches/patch-pine_init_c,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
--- pine/init.c.orig	Thu Apr  7 18:01:20 2005
+++ pine/init.c	Wed May  4 23:28:10 2005
@@ -1359,7 +1359,7 @@ init_pinerc(ps, debug_out)
     }
 
     if(!ps->pinerc){
-      build_path(buf, ps->home_dir, ".pinerc", sizeof(buf));
+      build_path(buf, ps->home_dir, ".etc/pinerc", sizeof(buf));
       ps->pinerc = cpystr(buf);
     }
 
@@ -3082,6 +3082,7 @@ process_feature_list(ps, list, old_growt
                              *lvalue[BM_SIZE * 8];
     int                       i,
                               yorn;
+    int                       bsmf = 1, bsmu = 1;
     long                      l;
     FEATURE_S		     *feat;
 
@@ -3103,6 +3104,12 @@ process_feature_list(ps, list, old_growt
     if(old_growth)
         set_old_growth_bits(ps, 0);
 
+    /* standards compliance */
+    if(bsmf)
+        F_TURN_ON(F_QUELL_FLOWED_TEXT, ps);
+    if(bsmu)
+	F_TURN_ON(F_COMPOSE_ALWAYS_DOWNGRADE, ps);
+
     /* now run through the list (global, user, and cmd_line lists are here) */
     if(list){
       for(p = list; (q = *p) != NULL; p++){
@@ -3165,7 +3172,7 @@ process_feature_list(ps, list, old_growt
       ps->orig_use_fkeys = 1;
 
     /* Will we have to build a new list? */
-    if(!(old_growth || hir || osr))
+    if(!(old_growth || hir || osr || bsmf || bsmu))
 	return;
 
     /*
@@ -3187,12 +3194,18 @@ process_feature_list(ps, list, old_growt
       }else if(old_growth && (strucmp(q, "old-growth") == 0 ||
                              strucmp(q, "no-old-growth") == 0)){
 	old_growth = 0;
+      }else if(bsmf && (strucmp(q, "quell-flowed-text") == 0 ||
+                             strucmp(q, "no-quell-flowed-text") == 0)){
+	bsmf = 0;
+      }else if(bsmu && (strucmp(q, "downgrade-multipart-to-text") == 0 ||
+                             strucmp(q, "no-downgrade-multipart-to-text") == 0)){
+	bsmu = 0;
       }
       lvalue[i++] = cpystr(q);
     }
 
     /* check to see if we still need to build a new list */
-    if(!(old_growth || hir || osr))
+    if(!(old_growth || hir || osr || bsmf || bsmu))
 	return;
 
     if(hir)
@@ -3201,6 +3214,10 @@ process_feature_list(ps, list, old_growt
       lvalue[i++] = "signature-at-bottom";
     if(old_growth)
       lvalue[i++] = "old-growth";
+    if(bsmf)
+      lvalue[i++] = "quell-flowed-text";
+    if(bsmu)
+      lvalue[i++] = "downgrade-multipart-to-text";
     lvalue[i] = NULL;
     set_variable_list(V_FEATURE_LIST, lvalue, TRUE, Main);
 }
@@ -6607,7 +6624,7 @@ char *filename;
     }
 #else	/* !DOS */
     if(!ps_global->pinerc){
-	build_path(buf, ps_global->home_dir, ".pinerc", sizeof(buf));
+	build_path(buf, ps_global->home_dir, ".etc/pinerc", sizeof(buf));
 	p = buf;
     }
 #endif	/* !DOS */
@@ -7528,7 +7545,7 @@ prune_folders(prune_cntxt, folder_base, 
     path2[sizeof(path2)-1] = '\0';
 
     if(F_ON(F_PRUNE_USES_ISO,ps_global)){             /* sent-mail-yyyy-mm */
-	sprintf(path2 + strlen(path2), "-%04.4d-%02.2d", month_to_use/12,
+	sprintf(path2 + strlen(path2), "-%4.4d-%2.2d", month_to_use/12,
 		month_to_use % 12 + 1);
     }
     else{
@@ -10213,7 +10230,7 @@ try_looking_in_stream:
 	/* if we got back a result from the status command, use it */
 	if(mm_status_result.flags){
 	    dprint(7, (debugfile,
-		   "rd_check_remvalid: got status_result 0x%x\n",
+		   "rd_check_remvalid: got status_result 0x%lX\n",
 		   mm_status_result.flags));
 	    if(mm_status_result.flags & SA_MESSAGES){
 		current_nmsgs = mm_status_result.messages;
@@ -10718,10 +10735,10 @@ try_wantto:
 	p += strlen(p);
 	sprintf(p, "Suspicious data in \"%.50s\": Continue anyway ",
 		(rd && rd->t.i.special_hdr) ? rd->t.i.special_hdr
-					    : unknown,
+					    : unknown/*,
 		(rd && rd->rn) ? rd->rn : unknown,
 		short_str((rd && rd->rn) ? rd->rn : "<noname>", tmp+1900,
-			  33, FrontDots));
+			  33, FrontDots)*/);
 	if(want_to(tmp, 'n', 'x', NO_HELP, WT_NORM) == 'y'){
 	    rv = 1;
 	    rd->flags |= USER_SAID_YES;
